sharedbuffers/mapped_struct.py

```diff
@@ -1875,6 +1877,124 @@ def _c_search_hkey_i32(hkey, pindex, stride0, length, hint):
                 return mid
         return lo
 
+    @cython.cfunc
+    @cython.locals(
+        hkey = cython.double,
+        mkey = cython.double,
+        lo = cython.size_t, hi = cython.size_t, length = cython.size_t,
+        mid = cython.size_t, mid2 = cython.size_t, stride0 = cython.size_t, hint = cython.size_t,
+        pindex = cython.p_char, skip = cython.size_t)
+    @cython.returns(cython.size_t)
+    def _c_search_hkey_f64(hkey, pindex, stride0, length, hint):
```

Comment:
The whole test suite tests these methods. There's an explicit test in testBsearchFloat64.

It's a binary search for doubles.